builtin/providers/azurerm/resource_arm_managed_disk.go

```diff
@@ -0,0 +1,256 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/arm/disk"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+	"net/http"
+	"strings"
+)
+
+func resourceArmManagedDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedDiskCreate,
+		Read:   resourceArmManagedDiskRead,
+		Update: resourceArmManagedDiskCreate,
+		Delete: resourceArmManagedDiskDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"storage_account_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(disk.PremiumLRS),
+					string(disk.StandardLRS),
+				}, true),
+			},
+
+			"create_option": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(disk.Import),
+					string(disk.Empty),
+					string(disk.Copy),
+				}, true),
+			},
+
+			"source_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"source_resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"os_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(disk.Windows),
+					string(disk.Linux),
+				}, true),
+			},
+
+			"disk_size_gb": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validateDiskSizeGB,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func validateDiskSizeGB(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(int)
+	if value < 1 || value > 1023 {
```

Comment:
I had actually just moved that function over from the vm resource (was pre-existing, just figured it made more sense to be based out of the new disk resource).
Also, according to some of the Microsoft folks I've been working with, the max size will be increasing to 4TB at some point hopefully soon.. Not sure when exactly, but I'll try and keep an eye out for when that happens and update this accordingly.